build/scripts/build-utils.ps1

```diff
@@ -312,15 +312,21 @@ function Get-MSBuildKindAndDir([switch]$xcopy = $false) {
         return
     }
 
-    # MSBuild from an active VS command prompt.  
+    # MSBuild from an active VS command prompt. Use the MSBuild here so long as it's from a 
+    # compatible Visual Studio. If not though throw and error out. Given the number of 
+    # environment variable changes in a developer command prompt it's hard to make guarantees
+    # about subbing in a new MSBuild instance
     if (${env:VSINSTALLDIR} -ne $null) {
         $command = (Get-Command msbuild -ErrorAction SilentlyContinue)
-        if ($command -ne $null) {
+        if ((Test-SupportedVisualStudioVersion ${env:VSCMD_VER}) -and ($command -ne $null) ) {
             $p = Split-Path -parent $command.Path
             Write-Output "vscmd"
             Write-Output $p
             return
         }
+        else {
+            throw "Developer Command Prompt for VS $(${env:VSCMD_VER}) is not recent enough. Please upgrade or build from a normal CMD window"
```

Comment:
Should this say what the minimum version should be?

Reply:
Considered that but seemed like it could be more confusing. You can't really upgrade to a specific version of VS, you just upgrade to latest. Hence why tell them a version they can't control?

Reply:
If we ever make a change that demands a prerelease version, you'd want to know that?

Reply:
"Please install 15.6.0-preview4." Not sure users are going to find that actionable over "Please install the latest preview of VS"

Reply:
It will when a user is trying to use preview 4 and we haven't even shipped that yet.